Factories/LoggerFactory.php

```diff
@@ -2,97 +2,90 @@
 
 namespace Infrastructure\Factories;
 
-
 use Infrastructure\Exceptions\InfrastructureException;
-use Infrastructure\Models\Logging\CloudWatchLoggingRegistry;
-use Infrastructure\Models\Logging\FileLoggingRegistry;
+use Psr\Log\LoggerInterface;
 
 class LoggerFactory
 {
     public const FILE = 'file';
     public const CLOUD_WATCH = 'cloudWatch';
+    public const SYSLOG = 'syslog';
+    public const ERROR_LOG = 'errorLog';
 
-    public const ALLOWED_LOGGER_TYPES = [
-        self::FILE,
-        self::CLOUD_WATCH,
-    ];
+    /**
+     * @var FileLogFactory
+     */
+    private $fileLogFactory;
 
     /**
-     * @var FileLoggingRegistry
+     * @var CloudWatchLogFactory
      */
-    private $fileLoggingRegistry;
+    private $cloudWatchLogFactory;
 
     /**
-     * @var CloudWatchLoggingRegistry
+     * @var ErrorLogFactory
      */
-    private $cloudWatchLoggingRegistry;
+    private $errorLogFactory;
 
     /**
-     * @var string
+     * @var SysLogFactory
      */
-    private $logPath;
+    private $sysLogFactory;
 
     /**
-     * @var string
+     * @var array
      */
-    private $applicationName;
+    private $loggers = [];
 
     /**
      * LoggerFactory constructor.
-     * @param string $logPath
-     * @param string $applicationName
+     * @param FileLogFactory $fileLogFactory
+     * @param CloudWatchLogFactory $cloudWatchLogFactory
+     * @param ErrorLogFactory $errorLogFactory
+     * @param SysLogFactory $sysLogFactory
      */
-    public function __construct(string $logPath, string $applicationName = '')
-    {
-        $this->fileLoggingRegistry = new FileLoggingRegistry();
-        $this->cloudWatchLoggingRegistry = new CloudWatchLoggingRegistry();
-        $this->logPath = $logPath;
-        $this->applicationName = $applicationName;
+    public function __construct(
```

Comment:
Think, that it can be easy to update, set array of loggers with keys in the constructor